mbzirc_ign/src/GameLogicPlugin.cc

```diff
@@ -0,0 +1,653 @@
+
+  /// \brief Event manager for pausing simulation
+  public: EventManager *eventManager{nullptr};
+
+  /// \brief Models with dead batteries.
+  public: std::set<std::string> deadBatteries;
+
+  /// \brief Amount of allowed setup time in seconds.
+  public: int setupTimeSec = 900;
+
+  /// \brief Mutex to protect the eventCounter.
+  public: std::mutex eventCounterMutex;
+
+  /// \brief Counter to create unique id for events
+  public: int eventCounter = 0;
+
+  /// \brief Total score.
+  public: double totalScore = 0.0;
+};
+
+//////////////////////////////////////////////////
+GameLogicPlugin::GameLogicPlugin()
+  : dataPtr(new GameLogicPluginPrivate)
+{
+}
+
+//////////////////////////////////////////////////
+GameLogicPlugin::~GameLogicPlugin()
+{
+  // set eventManager to nullptr so that Finish call does not attempt to
+  // pause sim
+  this->dataPtr->eventManager = nullptr;
+  this->dataPtr->Finish(this->dataPtr->simTime);
+}
+
+//////////////////////////////////////////////////
+void GameLogicPlugin::Configure(const ignition::gazebo::Entity & /*_entity*/,
+                           const std::shared_ptr<const sdf::Element> &_sdf,
+                           ignition::gazebo::EntityComponentManager & /*_ecm*/,
+                           ignition::gazebo::EventManager & _eventMgr)
+{
+  this->dataPtr->eventManager = &_eventMgr;
+
+  // Check if the game logic plugin has a <logging> element.
+  // example:
+  // <logging>
+  //   <path>/tmp</path>
+  // </logging>
+  const sdf::ElementPtr loggingElem =
+    const_cast<sdf::Element *>(_sdf.get())->GetElement("logging");
+
+  if (loggingElem)
+  {
+    // Get the logpath from the <path> element, if it exists.
+    if (loggingElem->HasElement("path"))
+    {
+      this->dataPtr->logPath =
+        loggingElem->Get<std::string>("path", "/dev/null").first;
+    }
+    else
+    {
+      // Make sure that we can access the HOME environment variable.
+      char *homePath = getenv("HOME");
+      if (!homePath)
+      {
+        ignerr << "Unable to get HOME environment variable. Report this error "
+          << "to https://github.com/osrf/mbzirc/issues/new. "
+          << "MBZIRC logging will be disabled.\n";
+      }
+      else
+      {
+        this->dataPtr->logPath = homePath;
+      }
+    }
+  }
+
+  ignmsg << "MBZIRC log path: " << this->dataPtr->logPath << std::endl;
+  common::removeAll(this->dataPtr->logPath);
+  common::createDirectories(this->dataPtr->logPath);
+
+  // Open the log file.
+  std::string filenamePrefix = "mbzirc";
+  this->dataPtr->logStream.open(
+      (common::joinPaths(this->dataPtr->logPath, filenamePrefix + "_" +
+      ignition::common::systemTimeISO() + ".log")).c_str(), std::ios::out);
+
+  // Open the event log file.
+  this->dataPtr->eventStream.open(
+      (common::joinPaths(this->dataPtr->logPath, "events.yml")).c_str(),
+      std::ios::out);
+
+  // Get the duration seconds.
+  if (_sdf->HasElement("run_duration_seconds"))
```

Comment:
The default value of `runDuration` is 0. Should we increase that value or show an error message if the parameter isn't set?

Reply:
increased to 1 hr / 3600s. 969f88f5112dbd6d8c6894787bae404723c50a2c